File: NX10.0.0.24/UGOPEN/SampleNXOpenApplications/.NET/CheckMate/ExecuteCheckerAndGetResults/ExecuteCheckerAndGetResults.cs
/*==================================================================================================

                 Copyright 2012 Siemens Product Lifecycle Management Software Inc.

                                      All Rights Reserved.

====================================================================================================
File description:

    NX/Open example to show how to create and execute journal to execute Checker and get its results
    by using NX Open C#.
====================================================================================================

*/

using System;
using NXOpen;

public class NXJournal
{
  public static void Main(string[] args)
  {
    // Get the NX session, work part, display part.
    Session theSession = Session.GetSession();
    Part workPart = theSession.Parts.Work;
    Part displayPart = theSession.Parts.Display;

    // Get the NX Check-Mate Validator object.
    NXOpen.Validate.Validator[] validators1;
    theSession.ValidationManager.FindValidator("Check-Mate", out validators1);

    // Get the NX Check-Mate ValidatorOptions, and set options.
    NXOpen.Validate.ValidatorOptions validatorOptions1;
    validatorOptions1 = validators1[0].ValidatorOptions;

    validatorOptions1.SkipChecking = false;
    validatorOptions1.SkipCheckingDontLoadPart = false;
    validatorOptions1.SaveResultInTeamcenter = NXOpen.Validate.ValidatorOptions.SaveModeTypes.DoNotSave;
    validatorOptions1.SavePartFile = NXOpen.Validate.ValidatorOptions.SaveModeTypes.DoNotSave;
    validatorOptions1.SaveResultInPart = false;

    // Clear part nodes if any.
    validators1[0].ClearPartNodes();

    // Appends the full path of a part file or the current work part.
    validators1[0].AppendPartNode(workPart);

    // Manually add this line to clean all existing tests
    validators1[0].ClearCheckerNodes();

    // Select a checker and append it into the Validator object.
    string[] classnames1 = new string[1];
    classnames1[0] = "%mqc_report_browseable_features";
    validators1[0].AppendCheckerNodes(classnames1);

    // Execute the Check-Mate checker.
    NXOpen.Validation.Result status1;
    status1 = validators1[0].Commit();

    // Get a Parser object, and show the checker result.
    NXOpen.Validate.Parser[] parsers1;
    theSession.ValidationManager.FindParser("Validation Gadget", out parsers1);
    
    parsers1[0].ClearResultObjects();
    parsers1[0].DataSource = NXOpen.Validate.Parser.DataSourceTypes.MostRecentRun;
    parsers1[0].MaxDisplayObjects = 10;
    parsers1[0].Commit();
  }
  public static int GetUnloadOption(string dummy) { return (int)Session.LibraryUnloadOption.Immediately; }
}

File: NX1899_NX1903_NX1907_NX1911_NX1915_NX1919/UGOPEN/SampleNXOpenApplications/Java/CheckMate/ExecuteCheckerAndGetResults/ExecuteCheckerAndGetResults.java
/*==================================================================================================

                 Copyright 2012 Siemens Product Lifecycle Management Software Inc.

                                      All Rights Reserved.

====================================================================================================
File description:

    NX/Open example to show how to create and execute journal to execute Checker and get its results
    by using NX Open Java.
====================================================================================================

*/


import nxopen.*;

public class ExecuteCheckerAndGetResults
{
  public static void main(String [] args) throws NXException, java.rmi.RemoteException
  {
    Session theSession = (Session)SessionFactory.get("Session");
    Part workPart = theSession.parts().work();
    Part displayPart = theSession.parts().display();

    nxopen.validate.Validator [] validators1 ;
    validators1 = theSession.validationManager().findValidator("Check-Mate");

    nxopen.validate.ValidatorOptions validatorOptions1;
    validatorOptions1 = validators1[0].validatorOptions();

    validatorOptions1.setSkipChecking(false);
    validatorOptions1.setSkipCheckingDontLoadPart(false);
    validatorOptions1.setSaveResultInTeamcenter(nxopen.validate.ValidatorOptions.SaveModeTypes.DO_NOT_SAVE);
    validatorOptions1.setSavePartFile(nxopen.validate.ValidatorOptions.SaveModeTypes.DO_NOT_SAVE);
    validatorOptions1.setSaveResultInPart(false);

    validators1[0].clearPartNodes();

    // Appends the full path of a part file or the current work part.
    validators1[0].appendPartNode(workPart);

    // Manually add this line to clean all existing tests
    validators1[0].clearCheckerNodes();

    String [] classnames1  = new String[1];
    classnames1[0] = "%mqc_report_browseable_features";
    validators1[0].appendCheckerNodes(classnames1);

    nxopen.Validation.Result status1;
    status1 = validators1[0].commit();

    nxopen.validate.Parser [] parsers1 ;
    parsers1 = theSession.validationManager().findParser("Validation Gadget");
    parsers1[0].clearResultObjects();
    parsers1[0].setDataSource(nxopen.validate.Parser.DataSourceTypes.MOST_RECENT_RUN);
    parsers1[0].setMaxDisplayObjects(10);
    parsers1[0].commit();
  }

  public static final int getUnloadOption() { return BaseSession.LibraryUnloadOption.IMMEDIATELY; }

}
